addons/purchase_stock/tests/test_purchase_delete_order.py

```diff
@@ -40,3 +40,53 @@ def test_00_delete_order(self):
         purchase_order_3.button_cancel()
         self.assertEqual(purchase_order_3.state, 'cancel', 'PO is cancelled!')
         purchase_order_3.unlink()
+
+    def test_01_delete_propagation(self):
+        ''' Testcase for deleting purchase order with linked move and propagate cancel off'''
+
+        partner = self.env['res.partner'].create({'name': 'My Partner'})
+        product = self.env['product.product'].create({
+            'name': 'Test',
+            'type': 'product',
+        })
+
+        stock_location = self.env.ref('stock.warehouse0').out_type_id.default_location_src_id
+        cust_location = self.env.ref('stock.stock_location_customers')
+        picking_type_out = self.ref('stock.picking_type_out')
+        picking_out = self.env['stock.picking'].create({
+            'location_id': stock_location.id,
+            'location_dest_id': cust_location.id,
+            'partner_id': partner.id,
+            'picking_type_id': picking_type_out,
+        })
+        move = self.env['stock.move'].create({
+            'name': product.name,
+            'product_id': product.id,
+            'product_uom_qty': 1,
+            'product_uom': product.uom_id.id,
+            'picking_id': picking_type_out,
+            'location_id': stock_location.id,
+            'location_dest_id': cust_location.id,
+        })
+        picking_out.action_confirm()
+
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': partner.id,
+            'order_line': [
+                (0, 0, {
+                    'product_id': product.id,
+                    'product_qty': 1.0,
+                    'product_uom': product.uom_id.id,
+                    'propagate_cancel': False,
+                })],
+        })
+        purchase_order.button_confirm()
+
+        qtys = [1]
+        in_ids = [purchase_order.order_line.move_ids.ids]
+        self.env['report.stock.report_reception'].action_assign(move.ids, qtys, in_ids)
+        purchase_order.button_cancel()
+        # Check purchase order and related move are canceled while linked move state is draft
+        self.assertEqual(purchase_order.state, 'cancel', 'Purchase Order should be canceled')
+        self.assertEqual(purchase_order.order_line.move_ids.state, 'cancel', 'Purchase order move should be canceled')
+        self.assertEqual(move.state, 'draft', 'Linked move should be draft')
```

Comment:
That assert is wrong considering what you're trying to test.
When you confirm your picking, `move` state will be `confirmed`, as there is no quantity in stock.
When you assign manually the reception to the delivery, moves will get linked, which means `move`'s state will be changed to `waiting`, as it's now waiting for its previous operation to finish.
Now, the cancelling of a move doesn't yet change the state of its next moves (it was added in `16.4` [here](https://github.com/odoo/odoo/pull/123759/files#diff-55c6314416a6a400da6acd5018d161a55eeeb0e3008fec8828121e3dd12be0ebR689-R690)). Instead, it will be still `waiting`.